libs/index/odc/index/stac.py

```diff
@@ -28,21 +30,48 @@
 }
 
 
+def _get_region_code(properties: Dict[str, Any]) -> str:
```

Comment:
My main concern with this approach is how to generalize/extend it. There are two modes here

1. Simple mode: some property if present maps to a string that contains region code
2. Landsat mode: several keys need to be present then some code needs to be supplied to form `region_code`

I would do something along these lines:
```python
candidates = ["odc:region_code", "s2:mgrs_tile", "io:supercell_id"]  # Extend by adding more candidates for new data sources
for rc_key in candidates:
   rc = properties.get(rc_key, None)
   if rc is not None:
       return rc
# do landsat later
```

```

Reply:
Yeah, I was thinking that too. I'll fix it up.